gufe/components/proteincomponent.py

```diff
@@ -211,25 +266,22 @@ def _from_openmmPDBFile(cls, openmm_PDBFile: PDBFile | PDBxFile, name: str = "")
         # Formal Charge
         netcharge = 0
         for a in rd_mol.GetAtoms():
+            atom_name = a.GetMonomerInfo().GetName().strip()
             atomic_num = a.GetAtomicNum()
-            atom_name = a.GetMonomerInfo().GetName()
 
             connectivity = sum(_BONDORDER_TO_ORDER[bond.GetBondType()] for bond in a.GetBonds())
             default_valence = periodicTable.GetDefaultValence(atomic_num)
 
-            if connectivity == 0:  # ions:
-                # strip catches cases like 'CL1' as name
-                if atom_name.strip(string.digits).upper() in positive_ions:
-                    fc = default_valence  # e.g. Sodium ions
-                elif atom_name.strip(string.digits).upper() in negative_ions:
-                    fc = -default_valence  # e.g. Chlorine ions
-                else:  # -no-cov-
+            if connectivity == 0:  # ions
+                ion_key = atom_name.strip().upper()
```

Comment:
Using upper here will give the wrong charge for ions that have multiple possible values which are determined by the name being all upper or a mix of upper and lower like `CE` and `Ce`.  Meaning we need to do a few different checks to cover all of the different possible input cases, maybe something like:
- is the unedited ion key in the list
- is the upper ion key in the list 
- is the upper digit stripped ion key in the list.
We should also add test cases which cover all of these checks.

Reply:
I agree and will add this!

Reply:
@jthorton how sure are you that we should be trying to match digit-stripped values? Do you think it could lead to confusion, or are cases like `CL1` pretty common that a user would expect it to be cast to `CL`?

Reply:
I've decided to not support digit stripped values in this PR  - with the idea of starting stricter so that we can be more lax in a follow-up PR if we chose.

Reply:
I don't think its too common but this is a regression as PDBs with CL1 that did work now won't. I am not sure how we make that clear to users or if we should still support those cases.

Reply:
Although I would normally be partial to your approach @atravitz, I think I agree with @jthorton on this case. Attempting to match digit stripped values would at least be a last resort here.

I think the changes in the PR are a welcome improvement, and reducing surprise from users for which the current behavior was working fine is worth us doing if possible.

Reply:
I agree with @jthorton as well - I didn't realize it would be a regression. Thanks josh!